src/Dropdown/Item.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { DropdownItem, IconContainer, Link, Text } from './Dropdown.style'
+import { DropdownItemProps } from './interface'
+
+const Item: React.FC<DropdownItemProps> = ({
+  isButton = 'button',
+  onClick = null,
+  href = '',
+  children = '',
+  icon = null,
+}) => {
+  if (isButton === 'button')
+    return (
+      <DropdownItem
+        onClick={() => {
+          if (onClick != null) onClick()
+        }}
+      >
+        <IconContainer>{icon}</IconContainer>
+        <Text>{children}</Text>
+      </DropdownItem>
+    )
+  else
+    return (
+      <Link href={href}>
+        <DropdownItem>
+          <IconContainer>{icon}</IconContainer>
+          <Text>{children}</Text>
+        </DropdownItem>
+      </Link>
+    )
```

Comment:
kayaknya styled component itu ada props `as`. [Lihat di sini deh](https://styled-components.com/docs/basics#extending-styles)

Jadi harusnya bisa begini kalo `isButton`nya diubah jadi boolean. Nanti style `Link` disatuin sama `DropdownItem` dan dibikin conditional aja 
```suggestion
   const props = {
    ...(onClick && isButton && { onClick }),
    ...(href && !isButton && { href }),
  }

  return (
    <DropdownItem as={isButton ? 'button' : 'a'} {...props}>
      <IconContainer>{icon}</IconContainer>
      <Text>{children}</Text>
    </DropdownItem>
  )
```